### DISCUSSION TITLE
Fall back to anonymous account when login expires

### DISCUSSION THREAD

**Discussion by brawaru** [Chatterino/chatterino2#3952] [Ideas]
It would make sense for Chatterino to automatically switch to anonymous account when login expires instead of spamming the chat with  ‘Login expired for user "{username}"! Try adding your account again.’ and trying to re-connect endlessly. Then you won't miss chat messages that were sent while your login was expired and don't need to restart client to see only part of them that was saved by history API.

**Comment by Felanbird** [comment]
> It would make sense for Chatterino to automatically switch to anonymous account when login expires instead of spamming the chat with ‘Login expired for user "{username}"! Try adding your account again.’

Unfortunately it wouldn't, as much as the message spam is annoying, the only thing switching you to anon would do is cause people to say `Chatterino logged me out of my account Madge`.

> Then you won't miss chat messages that were sent while your login was expired and don't need to restart client to see only part of them that was saved by history API.

This has already been resolved with #3878 - if you download and use the [nightly build](https://github.com/Chatterino/chatterino2/releases/tag/nightly-build), you will load message history after you re-login.
![image](https://user-images.githubusercontent.com/41973452/188887903-71e2b407-b66f-4785-b0d0-cc0b81714fef.png)

edit: I should note an approach for better-expired-notifications is being discussed in Twitch chat.

**Reply by brawaru** [reply]
> Unfortunately it wouldn't, as much as the message spam is annoying, the only thing switching you to anon would do is cause people to say Chatterino logged me out of my account Madge.

Well, the solution to those askers can be just cover the message field (or show red warning above it) with notice ‘Your login expired. Log in again to send messages’ (where latter has the link that opens accounts window).

It's more confusing to me that Chatterino tries to connect endlessly with expired credentials and floods the chat when it could've just continued with anonymous account. Maybe it can be an option for those who wouldn't expect this behaviour, e.g. ‘ ✅ Reconnect with anonymous account when login expires.’ Also, while refreshing chat history on account reconnect is good, those messages will still me missing from logs.

<p align="center">
<img src="https://user-images.githubusercontent.com/10401817/188895523-4e4e6102-93e3-43e9-869f-a6a76eadb24e.png" alt="Concept of expired login notice above the message">
</p>
